src/ansiblelint/transformer.py

```diff
@@ -0,0 +1,164 @@
+
+
+class FormattedEmitter(Emitter):
+    """Emitter that applies custom formatting rules when dumping YAML.
+
+    Root-level sequences are never indented.
+    All subsequent levels are indented as configured (normal ruamel.yaml behavior).
+
+    Earlier implementations used dedent on ruamel.yaml's dumped output,
+    but string magic like that had a ton of problematic edge cases.
+    """
+
+    _sequence_indent = 2
+    _sequence_dash_offset = 0  # Should be _sequence_indent - 2
+
+    # NB: mypy does not support overriding attributes with properties yet:
+    #     https://github.com/python/mypy/issues/4125
+    #     To silence we have to ignore[override] both the @property and the method.
+
+    @property  # type: ignore[override]
+    def best_sequence_indent(self) -> int:  # type: ignore[override]
+        """Return the configured sequence_indent or 2 for root level."""
+        return 2 if self.column < 2 else self._sequence_indent
+
+    @best_sequence_indent.setter
+    def best_sequence_indent(self, value: int) -> None:
+        """Configure how many columns to indent each sequence item (including the '-')."""
+        self._sequence_indent = value
+
+    @property  # type: ignore[override]
+    def sequence_dash_offset(self) -> int:  # type: ignore[override]
+        """Return the configured sequence_dash_offset or 2 for root level."""
+        return 0 if self.column < 2 else self._sequence_dash_offset
+
+    @sequence_dash_offset.setter
+    def sequence_dash_offset(self, value: int) -> None:
+        """Configure how many spaces to put before each sequence item's '-'."""
+        self._sequence_dash_offset = value
+
+
+# Transformer is for transforms like runner is for rules
+class Transformer:
+    """Transformer class performs the fmt transformations."""
+
+    def __init__(self, result: LintResult):
+        """Initialize a Transformer instance."""
+        # TODO: options for explict_start, indent_sequences
+        self.explicit_start = True
+        self.indent_sequences = True
+
+        self.matches: List[MatchError] = result.matches
+        self.files: Set[Lintable] = result.files
+
+        file: Lintable
+        lintables: Dict[str, Lintable] = {file.filename: file for file in result.files}
+        self.matches_per_file: Dict[Lintable, List[MatchError]] = {
+            file: [] for file in result.files
+        }
+
+        for match in self.matches:
+            try:
+                lintable = lintables[match.filename]
+            except KeyError:
+                # we shouldn't get here, but this is easy to recover from so do that.
+                lintable = Lintable(match.filename)
+                self.matches_per_file[lintable] = []
+            self.matches_per_file[lintable].append(match)
+
+    def run(self) -> None:
+        """Execute the fmt transforms."""
+        # ruamel.yaml rt=round trip (preserves comments while allowing for modification)
+        yaml = YAML(typ="rt")
```

Comment:
I can extract this into it's own function that configures ruamel.yaml. That should make the I/O of this method more obvious.

Reply:
I made a new `yaml` property to handle instantiating `ruamel.yaml.YAML` when needed.

Reply:
neat, looks good!